### PR TITLE
logpolicy: fix, test LogHost to work as documented

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

File: logpolicy/logpolicy.go [hunk 1]
```diff
@@ -74,7 +74,9 @@ func getLogTarget() string {
 // logtail server, or the default.
 func LogHost() string {
 	if v := getLogTarget(); v != "" {
-		return v
+		if u, err := url.Parse(v); err == nil {
```
Comment (andrew-d) [inline logpolicy/logpolicy.go:77]: Should we do this is in the `sync.Once` above?

**Review by andrew-d — APPROVED** [review]

File: logpolicy/logpolicy.go (hunk 1 above)
Comment (bradfitz) [reply logpolicy/logpolicy.go:77]: I considered it but there are like barely any callers and none performance sensitive so I erred on making a smaller diff

Comment (bradfitz) [reply logpolicy/logpolicy.go:77]: But if you feel strongly, you/I can in a follow-up?

Comment (andrew-d) [reply logpolicy/logpolicy.go:77]: Nah, just wanted to check 👍

File: logpolicy/logpolicy_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	}{
+		{"", "log.tailscale.io"},
+		{"http://foo.com", "foo.com"},
+		{"https://foo.com", "foo.com"},
+		{"https://foo.com/", "foo.com"},
+		{"https://foo.com:123/", "foo.com"},
+	}
+	for _, tt := range tests {
+		reset()
+		os.Setenv("TS_LOG_TARGET", tt.env)
```
Comment (maisem) [inline logpolicy/logpolicy_test.go:32]: t.Setenv

**Pull request merged**